packages/snap/src/index.ts

```diff
@@ -220,7 +220,10 @@ export const onRpcRequest = async ({
 
       const params = request.params as CreateIdentityRequestParams;
 
-      if (params?.privateKeyHex && !utils.isHexString(params?.privateKeyHex)) {
+      if (
+        params?.privateKeyHex &&
+        !utils.isHexString(`0x${params?.privateKeyHex}`)
```

Comment:
Won't it pass when `privateKeyHex` is an empty string?

Reply:
if there's no params?.privateKey, then it won't throw error. And then identity creation will use entropy

Reply:
But what if it's set to `''`? Could you check it manually, just to be sure?

Reply:
but, '' is a false value